chapter3.Rmd

```diff
@@ -542,29 +598,131 @@ test_object("non_us_some",
             incorrect_msg = "Did you assign to `non_us_some` the non-US revenue for the first two movies? You can use `[1:2,2]` to do this!")
 test_output_contains("mean(non_us_some)",
                      incorrect_msg = "Have you calculated the average of the values in `non_us_some` by calling `mean(non_us_some)`? Simply print out the result.")
-success_msg("Nice one! Continue to the next exercise.")
+success_msg("You are nearly at the end of the chapter. Stay on target!")
+```
+
+
+
+--- type:NormalExercise lang:r xp:100 skills:1 key:bf53de0964
+## Selection of matrix elements (2)
+
+For vectors, you learned about three other methods of indexing.
+
+- A vector of negative numbers selects elements to leave out.
+- A logical vector returns the elements where the index is `TRUE`.
+- A character vector returns elements where the element name matches the index.
+
+These indexing methods can also be used with vectors. This time you need two of them, and you can mix and match.
+
+```{r}
+a_matrix[-1, c(2, 4)] # everything except the first row, 2nd and 4th columns
+a_matrix[, c(FALSE, TRUE)] # every second column
+a_matrix[c("lightsaber", "blaster"), ] # rows named "lightsaber" and "blaster", all columns
```

Comment:
Given the limited horizontal real estate, this may look better if you place the comments above each line of code.